bin/pycbc_multi_inspiral

```diff
@@ -359,45 +447,52 @@ with ctx:
             sigma = {ifo : np.sqrt(sigmasq[ifo]) for ifo in args.instruments}
-            # Every time s_num is zero or we skip the segment, we run new 
+            # Every time s_num is zero or we skip the segment, we run new
             # template to increment the template index
             if s_num==0:
                 event_mgr.new_template(tmplt=template.params, sigmasq=sigmasq)
             if not analyse_segment: continue
             logging.info(
                 "Analyzing segment %d/%d", s_num + 1, len(segments[ifo]))
+            # The following dicts are created to store copies of the
+            # matched filtering results computed below
             snr_dict = dict.fromkeys(args.instruments)
             norm_dict = dict.fromkeys(args.instruments)
             corr_dict = dict.fromkeys(args.instruments)
             idx = dict.fromkeys(args.instruments)
             snrv_dict = dict.fromkeys(args.instruments)
             snr = dict.fromkeys(args.instruments)
-            # FIXME: 2 lines that can be moved outside the loops
-            #        We do not really use ifo_list, and we reassign
-            #        nifo identically a few lines below
-            ifo_list = args.instruments[:]
-            nifo = len(ifo_list)
             for ifo in args.instruments:
                 logging.info(
                     "Filtering template %d/%d, ifo %s", t_num + 1, n_bank, ifo)
                 # No clustering in the coherent search until the end.
                 # The correlation vector is the FFT of the snr (so inverse
-                # FFT it to get the snr). 
+                # FFT it to get the snr).
+              	# The following lines Unpack & store copies of the matched
+              	# filtering results for the current segment and template.
                 snr_ts, norm, corr, ind, snrv = \
                      matched_filter[ifo].matched_filter_and_cluster(
                          s_num, template.sigmasq(stilde[ifo].psd), window=0)
+                # snr time series for each ifo
                 snr_dict[ifo] = (
                         snr_ts[matched_filter[ifo].segments[s_num].analyze]
                         * norm)
                 norm_dict[ifo] = norm
+                # The correlation vector for each ifo. it is the  FFT of the snr
+                # (so inverse FFT it to get the snr).
                 corr_dict[ifo] = corr.copy()
+                # Trigger indices list for each ifo
                 idx[ifo] = ind.copy()
+                # Trigger snr list for each ifo
                 snrv_dict[ifo] = snrv.copy()
+		# Normalized trigger snr list for each ifo
                 snr[ifo] = snrv * norm
-            
-            # FIXME: wrong comment?
+
             # Move onto next segment if there are no triggers.
-            if len(ifo_list)==0: continue
+	    n_trigs = [len(snrv_dict[ifo]) for ifo in args.instruments]
+	    if np.all(n_trigs==np.zeros_like(args.instruments[:]), axis=0):
+		continue
```

Comment:
This is too complicated, how about this:
```suggestion
            n_trigs = [len(snrv_dict[ifo]) for ifo in args.instruments]
            if not any(n_trigs):
                continue
```
Also, be careful with tabs vs spaces. Generally we do not want tabs in Python code as they can have confusing side effects and misalign the indentation. You can configure your text editor to always use (four) spaces instead of a tab.

Reply:
Ah, in fact this seems to be causing the tests to fail:
```
[…]
TabError: inconsistent use of tabs and spaces in indentation
```

Reply:
@sebastiangomezlopez the tab/space issue seems to have been addressed, but my main point here still holds: the expression `np.all(n_trigs==np.zeros_like(args.instruments[:]), axis=0)` seems unnecessarily complicated. It seems that it is equivalent to `not any(n_trigs)`, right? If so, please use the simpler way.